libr/debug/p/native/linux/linux_debug.c

```diff
@@ -386,8 +391,8 @@ RDebugReasonType linux_dbg_wait(RDebug *dbg, int my_pid) {
 	// Ignore keyboard interrupt while waiting to avoid signaling the child process twice on
 	// break(SIGSTOP is sent by wait_break) which forced the user to continue an extra time
 	// to handle SIGINT
-	r_sys_signal (SIGINT, SIG_IGN);
-	r_cons_break_push ((RConsBreak)linux_dbg_wait_break, dbg);
```

Comment:
These wrappers were only replaced in continue+trace, keep in mind you have to implement it for stepping(even though it's a single step you may interrupt a system call, right now it breaks debug) and regular continue.
Also, I am still not convinced it had to be removed from here. And you will need to remove it from window/freebsd/macos's debug loop if you intend to have the interrupt wrap in debug.c.

Reply:
Right, this is not a complete fix to the real problem and the complete solution is in the multithreaded debugging PR.

Reply:
> And you will need to remove it from window/freebsd/macos's debug loop if you intend to have the interrupt wrap in debug.c.
The interrupt wrap already exists in `cmd_debug.c` and the callback is `static_debug_stop`. However, `r_sys_signal (SIGINT, SIG_IGN);` prevents that from working.

Reply:
Several commands have wrap the function with `static_debug_stop` to break the debugger out.
https://github.com/radareorg/radare2/blob/261783336c1114e796f8ec6b74149140f73bab10/libr/core/cmd_debug.c#L5170-L5175

Reply:
I have put them back and use `linux_dbg_wait_break` to send the `SIGINT` instead of ignoring and passing the `SIGINT` to the debuggee